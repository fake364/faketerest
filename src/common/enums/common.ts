export enum GENDER_ENUM {
  FEMALE = -1,
  NOT_SPECIFIED = 0,
  MALE
}
