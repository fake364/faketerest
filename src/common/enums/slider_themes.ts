export enum SLIDER_THEMES {
	DINNER_IDEA = 'dinnerIdea',
	HOME_DECOR = 'homeDecor',
	NEW_OUTFIT = 'newOutfit',
	THUMB_IDEA = 'thumbIdea'
}
