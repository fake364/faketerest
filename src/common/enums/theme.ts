export enum THEME_TYPE {
	BASE = 'base',
	BUSINESS = 'business',
	NEWSROOM = 'newsroom'
}