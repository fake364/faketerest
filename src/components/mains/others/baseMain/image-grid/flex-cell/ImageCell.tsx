import React from 'react';
import clsx from 'clsx';
import Image, { ImageProps } from 'next/image';
import { ShownType } from '../../../../../../common/types/common';
import { SLIDER_ARRAY } from '../../../../../../common/classes/customClasses/SliderNode';

interface Props extends ShownType {
  images: Array<ImageProps['src']>;
}

const ImageCell: React.FC<Props> = ({ images, shownType }) => {
  return (
    <div className="mx-2 my-2 h-[25vh] xl:h-[25vh] 2xl:h-[35vh] relative">
      {images.map((image, index) => {
        const isNotShown = shownType !== SLIDER_ARRAY[index];
        const animatedClasses = clsx(
          'duration-1000',
          isNotShown && 'invisible opacity-0'
        );
        return (
          <div className={animatedClasses}>
            <Image
              className={clsx('rounded-[16px]')}
              src={image}
              layout={'fill'}
              objectFit="cover"
            />
          </div>
        );
      })}
    </div>
  );
};

export default ImageCell;
