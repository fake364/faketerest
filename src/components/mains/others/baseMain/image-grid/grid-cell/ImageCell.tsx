import React from 'react';
import clsx from 'clsx';
import { rowStartClasses } from './utils/constants';
import Image from 'next/image';
import { ImageGridProps } from './ImageGridCell';
import { SLIDER_THEMES } from '../../../../../../common/enums/slider_themes';

interface Props extends Omit<ImageGridProps, 'isAnimated'> {
  style?: {
    WebkitTransformY: number;
    transformY: number;
  };
}

const sliderIndexMap = [
  SLIDER_THEMES.DINNER_IDEA,
  SLIDER_THEMES.HOME_DECOR,
  SLIDER_THEMES.NEW_OUTFIT,
  SLIDER_THEMES.THUMB_IDEA
];

const delays = [
  'delay-75',
  'delay-[500ms]',
  'delay-[700ms]',
  'delay-[900ms]',
  'delay-[1200ms]',
  'delay-[1500ms]',
  'delay-[1700ms]'
];

const notFirstRow = {
  0: 'delay-75',
  1: 'delay-[1700ms]',
  2: 'delay-[500ms]',
  3: 'delay-[1500ms]',
  4: 'delay-[700ms]',
  5: 'delay-[1200ms]',
  6: 'delay-[900ms]'
};

const ImageCell: React.FC<Props> = ({
  images,
  startFrom,
  index: animateIndex,
  shownType
}) => {
  return (
    <div
      className={clsx('row-span-4', rowStartClasses[startFrom - 1], 'relative')}
    >
      {images.map((image, index) => {
        const isNotShown = shownType !== sliderIndexMap[index];
        const chooseDelayIndex = animateIndex % 7;
        const delay = startFrom
          ? delays[chooseDelayIndex]
          : notFirstRow[chooseDelayIndex];
        const animatedClasses = clsx(
          'image-transitions',
          delay,
          isNotShown && 'invisible opacity-0'
        );
        return (
          <div key={sliderIndexMap[index]} className={animatedClasses}>
            <Image
              className={clsx('rounded-[16px]', 'h-max')}
              src={image}
              layout={'fill'}
              objectFit="cover"
            />
          </div>
        );
      })}
    </div>
  );
};

export default ImageCell;
