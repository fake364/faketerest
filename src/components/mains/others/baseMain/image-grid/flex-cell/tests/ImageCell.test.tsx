import React from 'react';
import { shallow } from 'enzyme';
import ImageCell from '../ImageCell';
import { SLIDER_THEMES } from '../../../../../../../common/enums/slider_themes';

describe('ImageCell', () => {
  it('should render three invisible images in one container and one visible depending on shown type', () => {
    const wrapper = shallow(
      <ImageCell
        images={['1', '2', '3', '4']}
        shownType={SLIDER_THEMES.HOME_DECOR}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
