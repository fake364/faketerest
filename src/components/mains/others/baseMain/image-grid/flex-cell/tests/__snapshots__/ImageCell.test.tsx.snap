// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCell should render three invisible images in one container and one visible depending on shown type 1`] = `
<div
  className="mx-2 my-2 h-[25vh] xl:h-[25vh] 2xl:h-[35vh] relative"
>
  <div
    className="duration-1000 invisible opacity-0"
  >
    <Image
      className="rounded-[16px]"
      height={500}
      layout="fill"
      objectFit="cover"
      quality={70}
      sizes="12vw"
      src="1"
      width={200}
    />
  </div>
  <div
    className="duration-1000"
  >
    <Image
      className="rounded-[16px]"
      height={500}
      layout="fill"
      objectFit="cover"
      quality={70}
      sizes="12vw"
      src="2"
      width={200}
    />
  </div>
  <div
    className="duration-1000 invisible opacity-0"
  >
    <Image
      className="rounded-[16px]"
      height={500}
      layout="fill"
      objectFit="cover"
      quality={70}
      sizes="12vw"
      src="3"
      width={200}
    />
  </div>
  <div
    className="duration-1000 invisible opacity-0"
  >
    <Image
      className="rounded-[16px]"
      height={500}
      layout="fill"
      objectFit="cover"
      quality={70}
      sizes="12vw"
      src="4"
      width={200}
    />
  </div>
</div>
`;
